### PR TITLE
Add upgrade storybook documentation

### PR DESCRIPTION
**Pull request by GeoffCoxMSFT**
This moves the upgrade guide from the fluentui-react-upgrade repository into the storybook documentation.
Starting as a draft PR to get an initial review of the content.

- rewrote several portions to be target to public consumers instead of internal partners.
- added an upgrade folder at the same level as developer, put a 'from v8' subfolder to allow for future 'from v0'.
- this isn't the end-all-be-all of the upgrade documentation, so appreciate comments on what else needs to be added later.
- not sure yet where shims and code-mods will end up, discussing with @jurokapsiar.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |         Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `175.69 kB`<br />`48.974 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            | `32.526 kB`<br />`10.645 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/31065b920170a168f3aee19cfdd83530cc1fb7dc'>31065b920170a168f3aee19cfdd83530cc1fb7dc</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22093/perf-test-react-components/Avatar-mount_base.html">1248</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22093/perf-test-react-components/Avatar-mount.html">1304</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22093/perf-test-react-components/Button-mount_base.html">746</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22093/perf-test-react-components/Button-mount.html">765</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22093/perf-test-react-components/FluentProvider-mount_base.html">2199</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22093/perf-test-react-components/FluentProvider-mount.html">2195</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22093/perf-test-react-components/FluentProviderWithTheme-mount_base.html">353</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22093/perf-test-react-components/FluentProviderWithTheme-mount.html">391</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22093/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">333</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22093/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">327</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22093/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">423</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22093/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">421</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22093/perf-test-react-components/MakeStyles-mount_base.html">1952</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22093/perf-test-react-components/MakeStyles-mount.html">1943</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/src/Concepts/Upgrade/FromV8/Planning.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+You should plan a set of getting started tasks for upgrading one component in one location.
+
+1. Update project to reference Fluent React v9.
+2. Add FluentProvider with a theme to the root of the app
+3. Add any v9 component to the UI, verify it renders as expected, then remove it.
+4. Update a single usage location to use the v9 component. Consider recording how long it takes.
+
+### Iterative planning
+
+Plan a tasks or set of tasks to be able to 'rinse and repeat' each usage upgrade.
```
Comment (khmakoto) [inline packages/react-components/src/Concepts/Upgrade/FromV8/Planning.stories.mdx:178]: _nit:_

```suggestion
Plan a task or set of tasks to be able to 'rinse and repeat' each usage upgrade.
```

File: packages/react-components/src/Concepts/Upgrade/FromV8/Troubleshooting.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { Meta } from '@storybook/addon-docs';
+
+<Meta title="Concepts/Upgrading/from v8/Troubleshooting" />
+
+## "I tried using the new components but there does not seem to be any styling applied to them. Am I doing something wrong?"
+
+Components in version 8 could be used in isolation and still retain their default styling
```
Comment (khmakoto) [inline packages/react-components/src/Concepts/Upgrade/FromV8/Troubleshooting.stories.mdx:7]: nit: Missing dot

```suggestion
Components in version 8 could be used in isolation and still retain their default styling.
```

File: packages/react-components/src/Concepts/Upgrade/FromV8/Troubleshooting.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+## "I managed to get the theme working but the components look different than they did previously, why is that?"
+
+We are modernizing our components in version 9 to adhere to the latest guidelines in the Fluent Design Language. That is the reason for any visual differences you might notice.
+
+## "What if I want to override a component's styles? Say, if I want to make them look like they did before? It seems like the styles prop does not exist anymore!"
+
+The styling story in version 9 is very different to what existed in version 8. For one, we are moving away from providing a `styles` prop and are mostly focusing of providing styling via the `className` prop (if you are worried about how could you style every piece of a component via `className`, do not fret and keep reading).
+
+We are also moving away from `mergeStyles` in favor of `makeStyles`, another in-house CSS-in-JS solution inspired that allows for things like atomic classes and build-time optimization of styles.
```
Comment (khmakoto) [inline packages/react-components/src/Concepts/Upgrade/FromV8/Troubleshooting.stories.mdx:35]:
```suggestion
We are also moving away from `mergeStyles` in favor of `makeStyles` from [`@griffel/react`](https://github.com/microsoft/griffel), a new in-house CSS-in-JS solution that allows for things like atomic classes and build-time optimization of styles.
```

**Review by khmakoto — APPROVED** [review]

File: packages/react-components/src/Concepts/Upgrade/FromV8/ImportantChanges.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    <Button styles={customButtonStyles}>Hello Custom Styles</Button>
+  </ThemeProvider>
+);
+```
+
+### v9
+
+In v9, styles are created with the `makeStyles` function and combined with the `mergeClasses` function.
+Components apply styles through the `className` property.
+The styles are created at build time, but never runtime.
```
[the remark this answers is not in the document]
Comment (GeoffCoxMSFT) [reply packages/react-components/src/Concepts/Upgrade/FromV8/ImportantChanges.stories.mdx]: We definitely need a lot more docs on how to integrate this into a fluent-v9 based app.

File: packages/react-components/src/Concepts/Upgrade/FromV8/Overview.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+## What is available in v9?
+
+Version 9 is a "converged" library built from the ground-up that addresses many of the concerns and issues that plagued version 8.
+However, this approach means that the intial number of version 9 components is fewer than existed in version 8.
+Some components have also been renamed and a couple of component retired.
+
+At the time of this writing, v9 has an initial set of components shipped as a release candidate.
+There are also some components in preview that are more likely to change than the release candidate components.
+However, **all published components are production ready**.
```
[the remark this answers is not in the document]
Comment (GeoffCoxMSFT) [reply packages/react-components/src/Concepts/Upgrade/FromV8/Overview.stories.mdx:52]: Previously chatted with Justin about this and our stance is that the published components are production ready. They may have API changes later depending on their alpha, beta, RC, shipped status.  I'll add a link out to the summary of the label meanings as part of a later PR.

**Pull request merged**